### PR TITLE
feat(volo-build, volo-cli): refactor config model and correspond commands

### PR DESCRIPTION
**Pull request by Ggiggle**
## Motivation
For the usability of config file, the same git repo for different idls should be managed as one source, and another goal is to reuse the common part of single config and workspace config for better coding.

## Solution
1. Refactor the config model to manage repo unitedly, and extract the common part between the single and workspace config.
2. Support the repo command, including add and update cmd for users manage their repos by cli.
3. Adapt the idl and init command to the new model.

### CODE REVIEW

**Review by PureWhiteWu — DISMISSED** [review]

File: volo-cli/src/repo/add.rs [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+        help = "Specify the git repo name for repo.\nExample: cloudwego_volo"
+    )]
+    pub repo: Option<String>,
+    #[arg(
+        short = 'g',
+        long = "git",
+        help = "Specify the git repo for idl.\nShould be in the format of \
+                \"git@domain:path/repo.git\".\nExample: git@github.com:cloudwego/volo.git"
+    )]
+    pub git: Option<String>,
```
[the remark this answers is not in the document]
Comment (Ggiggle) [reply volo-cli/src/repo/add.rs:24]: git 是必需的，我改一下，repo 实现了从 git url 里提取仓库名的函数，可以不传

File: volo-build/src/legacy/model.rs [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+#[derive(Serialize, Deserialize, Debug, Clone)]
+pub struct Entry {
+    pub protocol: IdlProtocol,
+    pub filename: PathBuf,
+
+    pub idls: Vec<Idl>,
+    #[serde(default)]
+    pub touch_all: bool,
+    #[serde(default)]
+    pub nonstandard_snake_case: bool,
```
Comment (Millione) [inline volo-build/src/legacy/model.rs:22]: 漏掉了

File: volo-cli/src/init.rs [hunk 3]
```diff
@@ -114,7 +120,6 @@ impl Init {
     }
 
     fn copy_thrift_template(&self, config_entry: Entry) -> anyhow::Result<()> {
-        std::env::set_var("OUT_DIR", "/tmp/idl");
```
[the remark this answers is not in the document]
Comment (Ggiggle) [reply volo-cli/src/init.rs:117]: 用了 [tempdir](https://docs.rs/tempdir/latest/tempdir/) crate，出了 init 函数的作用域会自动删除声明的 tempdir

File: volo-cli/README.md [hunk 4]
```diff
@@ -27,6 +27,8 @@ SUBCOMMANDS:
     help    Print this message or the help of the given subcommand(s)
     idl     manage your idl
```
[the remark this answers is not in the document]
Comment (Ggiggle) [reply volo-cli/README.md:28]: 这里我改成只保留最新版的命令

File: volo-cli/src/init.rs [hunk 5]
```diff
@@ -28,14 +36,13 @@ pub struct Init {
...
     pub r#ref: Option<String>,
-
     #[arg(
         short = 'i',
         long = "includes",
         help = "Specify the include dirs for idl.\nIf -g or --git is specified, then this should \
                 be the path in the specified git repo."
     )]
-    pub includes: Option<Vec<PathBuf>>,
+    pub includes: Vec<PathBuf>,
```
[the remark this answers is not in the document]
Comment (Ggiggle) [reply volo-cli/src/init.rs:45]: 测试的时候发现不需要用 option 表示是否是可选的，为了减少编码的时候的option的判断，就都去掉了

File: volo-cli/src/model.rs [hunk 6]
```diff
@@ -2,9 +2,18 @@ use anyhow::Result;
...
+use crate::{
+    command::CliCommand, context::Context, http::Http, idl::Idl, init::Init, migrate::Migrate,
+    repo::Repo,
+};
 
-define_commands!(Subcommand { Init, Idl, Http });
+define_commands!(Subcommand {
+    Init,
+    Repo,
+    Idl,
```
[the remark this answers is not in the document]
Comment (Ggiggle) [reply volo-cli/src/model.rs:13]: 还需要的，用户使用上保持兼容，另外idl和repo都有add命令

File: volo-build/src/legacy/model.rs (hunk 2 above)
Comment (Ggiggle) [reply volo-build/src/legacy/model.rs:22]: 这里是原本的model，要不就不改了，只是为了做迁移的

**Review by PureWhiteWu — DISMISSED** [review]

**Review by PureWhiteWu — APPROVED** [review]

**Review by Millione — APPROVED** [review]

**Pull request merged**